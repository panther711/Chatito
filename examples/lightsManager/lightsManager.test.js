const fs = require("fs");
const path = require("path");
const generator = require("../../core/datasetGenerator");
const cp = require( "child_process" );

const getExampleFile = filename => path.resolve(__dirname, filename);

test("test dataset generation", () => {
    const fileContent = fs.readFileSync(getExampleFile("lightsManager.chatito"), "utf8");
    const result = generator.datasetFromString(fileContent);
    expect(JSON.stringify(result, null, 2)).toMatchSnapshot();
});

test("test npm command line generator for snips dataset", () => {
    const d = __dirname;
    const generatedFile = path.resolve(`${d}/lightsManager_snips_training_100.json`);
    const npmBin = path.resolve(`${d}/../../bin/chatito.js`);
    const grammarFile = path.resolve(`${d}/lightsManager.chatito`);
    const args = "--format=snips --max=100 --min=100";
    if (fs.existsSync(generatedFile)) { fs.unlinkSync(generatedFile); }
    const child = cp.execSync(`node ${npmBin} ${grammarFile} ${args}`);
    expect(child.toString("utf8")).toMatchSnapshot();
    expect(fs.existsSync(generatedFile)).toBeTruthy();
    fs.unlinkSync(generatedFile);
    expect(fs.existsSync(generatedFile)).toBeFalsy();
});

test("test npm command line generator for rasa dataset", () => {
    const d = __dirname;
    const generatedFile = path.resolve(`${d}/lightsManager_rasa_training_100.json`);
    const npmBin = path.resolve(`${d}/../../bin/chatito.js`);
    const grammarFile = path.resolve(`${d}/lightsManager.chatito`);
    const args = "--format=rasa --max=100 --min=100";
    if (fs.existsSync(generatedFile)) { fs.unlinkSync(generatedFile); }
    const child = cp.execSync(`node ${npmBin} ${grammarFile} ${args}`);
    expect(child.toString("utf8")).toMatchSnapshot();
    expect(fs.existsSync(generatedFile)).toBeTruthy();
    fs.unlinkSync(generatedFile);
    expect(fs.existsSync(generatedFile)).toBeFalsy();
});