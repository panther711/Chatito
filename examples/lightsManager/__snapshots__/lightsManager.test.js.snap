// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dataset generation 1`] = `
"{
  \\"rasa_nlu_data\\": {
    \\"regex_features\\": [],
    \\"entity_synonyms\\": [
      {
        \\"value\\": \\"off\\",
        \\"synonyms\\": [
          \\"turn off\\",
          \\"deactivate\\"
        ]
      },
      {
        \\"value\\": \\"on\\",
        \\"synonyms\\": [
          \\"turn on\\",
          \\"activate\\"
        ]
      }
    ],
    \\"common_examples\\": [
      {
        \\"text\\": \\"Hey Bot turn off the lights\\",
        \\"intent\\": \\"lightChange\\",
        \\"entities\\": [
          {
            \\"start\\": 8,
            \\"end\\": 16,
            \\"value\\": \\"off\\",
            \\"entity\\": \\"switch\\"
          },
          {
            \\"start\\": 21,
            \\"end\\": 27,
            \\"value\\": \\"lights\\",
            \\"entity\\": \\"lights\\"
          }
        ]
      },
      {
        \\"text\\": \\"Hey Bot turn off the lowLights\\",
        \\"intent\\": \\"lightChange\\",
        \\"entities\\": [
          {
            \\"start\\": 8,
            \\"end\\": 16,
            \\"value\\": \\"off\\",
            \\"entity\\": \\"switch\\"
          },
          {
            \\"start\\": 21,
            \\"end\\": 30,
            \\"value\\": \\"lowLights\\",
            \\"entity\\": \\"lights\\"
          }
        ]
      },
      {
        \\"text\\": \\"Hey Bot deactivate the lights\\",
        \\"intent\\": \\"lightChange\\",
        \\"entities\\": [
          {
            \\"start\\": 8,
            \\"end\\": 18,
            \\"value\\": \\"off\\",
            \\"entity\\": \\"switch\\"
          },
          {
            \\"start\\": 23,
            \\"end\\": 29,
            \\"value\\": \\"lights\\",
            \\"entity\\": \\"lights\\"
          }
        ]
      },
      {
        \\"text\\": \\"Hey Bot deactivate the lowLights\\",
        \\"intent\\": \\"lightChange\\",
        \\"entities\\": [
          {
            \\"start\\": 8,
            \\"end\\": 18,
            \\"value\\": \\"off\\",
            \\"entity\\": \\"switch\\"
          },
          {
            \\"start\\": 23,
            \\"end\\": 32,
            \\"value\\": \\"lowLights\\",
            \\"entity\\": \\"lights\\"
          }
        ]
      },
      {
        \\"text\\": \\"Hey Bot turn on the lights\\",
        \\"intent\\": \\"lightChange\\",
        \\"entities\\": [
          {
            \\"start\\": 8,
            \\"end\\": 15,
            \\"value\\": \\"on\\",
            \\"entity\\": \\"switch\\"
          },
          {
            \\"start\\": 20,
            \\"end\\": 26,
            \\"value\\": \\"lights\\",
            \\"entity\\": \\"lights\\"
          }
        ]
      },
      {
        \\"text\\": \\"Hey Bot turn on the lowLights\\",
        \\"intent\\": \\"lightChange\\",
        \\"entities\\": [
          {
            \\"start\\": 8,
            \\"end\\": 15,
            \\"value\\": \\"on\\",
            \\"entity\\": \\"switch\\"
          },
          {
            \\"start\\": 20,
            \\"end\\": 29,
            \\"value\\": \\"lowLights\\",
            \\"entity\\": \\"lights\\"
          }
        ]
      },
      {
        \\"text\\": \\"Hey Bot activate the lights\\",
        \\"intent\\": \\"lightChange\\",
        \\"entities\\": [
          {
            \\"start\\": 8,
            \\"end\\": 16,
            \\"value\\": \\"on\\",
            \\"entity\\": \\"switch\\"
          },
          {
            \\"start\\": 21,
            \\"end\\": 27,
            \\"value\\": \\"lights\\",
            \\"entity\\": \\"lights\\"
          }
        ]
      },
      {
        \\"text\\": \\"Hey Bot activate the lowLights\\",
        \\"intent\\": \\"lightChange\\",
        \\"entities\\": [
          {
            \\"start\\": 8,
            \\"end\\": 16,
            \\"value\\": \\"on\\",
            \\"entity\\": \\"switch\\"
          },
          {
            \\"start\\": 21,
            \\"end\\": 30,
            \\"value\\": \\"lowLights\\",
            \\"entity\\": \\"lights\\"
          }
        ]
      }
    ]
  }
}"
`;

exports[`test npm command line generator for rasa dataset 1`] = `
"/Users/rodrigo/Trabajo/Chatito/examples/lightsManager/lightsManager_rasa_training_100.json
{ total: 8, training: 100, testing: 0 }
"
`;

exports[`test npm command line generator for snips dataset 1`] = `
"/Users/rodrigo/Trabajo/Chatito/examples/lightsManager/lightsManager_snips_training_100.json
{ total: 8, training: 100, testing: 0 }
"
`;
